### ISSUE TITLE
LazyInitializationException during startup if using repo.getById, but not repo.findById

### ISSUE THREAD

**Issue by chrisxaustin** [spring-projects/spring-data-jpa#2336]
Spring Boot 2.5.5

Original issue: https://github.com/spring-projects/spring-boot/issues/28309
Related: https://github.com/spring-projects/spring-data-jpa/issues/2261

In 2.4.4 I was able to use either `findById` or `getById`, but in 2.5.5 only `findById` returns an object that I can safely use.
The object returned by `getById` will throw `LazyInitializationException` with the message "No Session" if I try to access any attributes, e.g. name in this example:

```java
@Repository
public interface FooRepository extends CrudRepository<Foo, Long> {

    Optional<Foo> findById(Long aLong);

    Foo getById(long id);

}

@Data
@Entity
public class Foo {

    @Id
    private long id;

    private String name;

}
```

Sample service accessing the repo 
```java
@Service
@RequiredArgsConstructor
@Slf4j
public class FailService {

    private final FooRepository fooRepo;

    @PostConstruct
    public void init() {
        Foo f = fooRepo.findById(1L).get();
        log.info("findById foo.id: " + f.getId());
        log.info("findById foo.name: " + f.getName());

        f = fooRepo.getById(1);
        log.info("getById foo.id: " + f.getId());
        log.info("getById foo.name: " + f.getName());
    }
}
```

The first three lines print, the final one throws

```
org.hibernate.LazyInitializationException: could not initialize proxy [com.example.demo.Foo#1] - no Session
```

Minimal case attached with a junit test that demonstrates the problem.

[getById-regression.zip](https://github.com/spring-projects/spring-data-jpa/files/7347646/getById-regression.zip)

**Comment by philwebb** [comment]
@chrisxaustin I totally agree that this should have been documented, but I don't think we should do that in the Spring Boot release notes. We have so many dependencies that it would be very hard for us to include the API changes of each one.

**Comment by chrisxaustin** [comment]
A list of links to the breaking changes for common components that were updated would be sufficient. A new Spring Boot user shouldn't be expected to know which project versions they need to check.

The change seems very reasonable and consistent btw - I see this as a documentation/usability bug that could be avoided. It could also be solved by using a more helpful error.

 I went down too many unproductive paths while trying to troubleshoot the issue - my initial hypothesis was that the proxy was not fully initialized when I accessed the repository from a @PostConstruct method, which led to experimenting with EventListener and other approaches before I realized that it only occurred on the getById paths. I couldn't find anything in the Spring issues, stackoverflow, etc.
